### DISCUSSION TITLE
modifying shortcuts for CmdNextTab

### DISCUSSION THREAD

**Discussion by ghost** [sumatrapdfreader/sumatrapdf#2797] [Q&A] [answered]
From the list of shortcut commands, I would like to map "CmdNextTab" and "CmdPrevTab" onto the Ctrl + Page Up and Ctrl + Page Down commands. I am using the following lines in the SumatraPDF-settings.txt file. However, neither command is responsive. Any suggestions? Thank you. 

Shortcuts [
	[
		Cmd = CmdNextTab
		Key = Ctrl + Page Up
	]
	[
		Cmd = CmdPrevTab
		Key = Ctrl + Page Down
	]
	[
		Cmd = CmdOpenFolder
		Key = Alt + O
	]
]

**Comment by GitHubRulesOK** [comment, marked as answer]
Ensure your using the 3.5 pre-release or similar
 your last one works for me  no problem (Cmd = CmdOpenFolder Key = Alt + O)

However there are two issues with your tabs assignments
First remove the spaces its `PageUp` and `PageDown` for those single keys
However on my keyboard I manually need to use Ctrl with Fn to call PageUp and PageDown so beware laptop layouts can be different

**Reply by ghost** [reply]
Thank you. I was running 3.4 and after updating to the 3.5 pre-release, it works.

**Reply by GitHubRulesOK** [reply]
It should have worked with 3.4 BUT most likely the spaces after Page were your main issue.

**Comment by mjmatthiesen** [comment]
Just as an aside, the ctrl+pgdn/pgup are default windows behaviour and should be supported on top of anything else a user would want to configure.

**Reply by GitHubRulesOK** [reply]
@mjmatthiesen those keys are app specific to a document viewer thus different for modern one page browsers where each page is horiontally tabbed since there is no concept of continuous "pages" so SumatraPDF used the same convention as other doc viewers like Word, Acrobat etc

![image](https://user-images.githubusercontent.com/12754913/188243363-7621b1da-6670-470a-bbf6-9703a3b0fdd8.png)
![image](https://user-images.githubusercontent.com/12754913/188243749-57fefebf-7ee8-4ca6-b829-58fc37b5a27e.png)

**Comment by rfpozo** [comment]
Hi,
did you solve the problem for the CmdPrevTab/CmdNextTab implementation? I'm running 3.5 and still doesn't work for me.
Thanks!

**Reply by ghost** [reply]
Hello, 
Yes, it works for me - I am using v3.5.15239, 64-bit, pre-release. Here is the code snippet I am using: 


Shortcuts [
	[
		Cmd = CmdPrevTab
		Key = Ctrl + PageUp
	]
	[
		Cmd = CmdNextTab
		Key = Ctrl + PageDown
	]
	[
		Cmd = CmdNone
		Key = q
	]
]


Hope this works!

**Reply by rfpozo** [reply]
> 

Installing the pre release solved my problem. Thank you!!

**Reply by GitHubRulesOK** [reply]
@rfpozo just beware that this (last) month pre-rel is with annotation in file a potential crash
Most stable with command line support is currently 3.4.5

**Comment by flyboy101** [comment]
"CmdNextTab" and "CmdPrevTab" are not working on Release v3.4.5 64-bit as per the following snippet:
[
Cmd = CmdNextTab
Key = Ctrl + Right
]
[
Cmd = CmdPrevTab
Key = Ctrl + Left
]

**Reply by GitHubRulesOK** [reply]
The commands are constantly expanding over time so not all were available in 3.4.0 however the more basic should work to map a new combination.

in my current copy
```
Shortcuts [
	[
		Cmd = CmdCloseTabsToTheRight
		Key = Alt + o
	]
[
Cmd = CmdNextTab
Key = Ctrl + Right
]
[
Cmd = CmdPrevTab
Key = Ctrl + Left
]
]
```
seems to be working ensure the group has its own ]

**Reply by flyboy101** [reply]
I do confirm the default combinations work as intended [1], i.e.

`Ctrl + Tab` - go to next tab
`Ctrl + Shift + Tab` - go to previous tab

The commands IDs "CmdNextTab" and "CmdPrevTab" are not listed in the full list [2], but they seem available on the file `Commands.h` [3], though.

Which release are you using on your machine?

References:
[1] https://www.sumatrapdfreader.org/docs/Keyboard-shortcuts
[2] https://www.sumatrapdfreader.org/docs/Commands
[3] https://github.com/sumatrapdfreader/sumatrapdf/blob/master/src/Commands.h#L9

**Reply by GitHubRulesOK** [reply]
I am using last but one pre-release 3.5.15262 (current 264 has a couple of new issues) to download it should be go to download and edit shortcut to one less. https://www.sumatrapdfreader.org/prerelease
https://files.sumatrapdfreader.org/file/kjk-files/software/sumatrapdf/prerel/15263/SumatraPDF-prerel-64.exe
